Index: kwallet-framework-4.95.0/src/CMakeLists.txt
===================================================================
--- kwallet-framework-4.95.0.orig/src/CMakeLists.txt	2014-01-03 12:00:34.000000000 +0000
+++ kwallet-framework-4.95.0/src/CMakeLists.txt	2014-01-22 12:59:20.268332298 +0000
@@ -41,7 +41,7 @@
     target_link_libraries(KF5Wallet PRIVATE ${CARBON_LIBRARY})
 else()
     target_link_libraries(KF5Wallet PRIVATE Qt5::DBus)
-    install(FILES ${kwallet_xml} DESTINATION ${DBUS_INTERFACES_INSTALL_DIR})
+    install(FILES ${kwallet_xml} DESTINATION ${DBUS_INTERFACES_INSTALL_DIR} RENAME kf5_org.kde.KWallet.xml)
 endif()
 
 set_target_properties(KF5Wallet PROPERTIES VERSION   ${KWALLET_VERSION_STRING}
